to-do-app/src/components/UpcomingActivities.js

```diff
@@ -0,0 +1,37 @@
+import {useState,useEffect} from 'react'
+const UpcomingActivities=()=>{
+    let today = new Date();
+    let currentDate=today.toJSON().slice(0,10);
+        const [tasks,setTasks]=useState([]);
+        const compareDates=(t1,t2)=>t1.date.localeCompare(t2.date);
+        useEffect(()=>{
+            const data=JSON.parse(localStorage.getItem("tasks"));
+            if(data){
+                setTasks(data);
+            }
+        },[])
+        
+        useEffect(()=>{
+          localStorage.setItem("tasks",JSON.stringify(tasks));
+        },[tasks])
+    return(
+        <div className="items">
+            <table className="table">
+                <tbody>
+                    {tasks.filter((task)=>task.date.slice(0,10)!==currentDate).sort(compareDates).map((task)=>{
```

Comment:
Is not clear why date is sliced here.
Maybe move this in a function that explains what's going on